### PR TITLE
Adding various POJOS

### PR DESCRIPTION
**Pull request by SKeeneCode**
Adding POJOS for Leaderboard, MapPool, MatchmakerQueue and MatchmakerQueueMapPool

### CODE REVIEW

File: faf-commons-api/src/main/java/com/faforever/commons/api/dto/MatchmakerQueue.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import com.github.jasminb.jsonapi.annotations.Relationship;
+import com.github.jasminb.jsonapi.annotations.Type;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+@Type("matchmakerQueue")
+public class MatchmakerQueue extends AbstractEntity {
```
Comment (Brutus5000) [inline faf-commons-api/src/main/java/com/faforever/commons/api/dto/MatchmakerQueue.java:13]: Maye add `matchmakerQueueMapPool` relationship

**Review by Brutus5000 — CHANGES_REQUESTED** [review]
We need `@EqualsAndHashCode(callSuper = true)` annotated on all classes

**Comment by bukajsytlos** [comment]
didn't we move this to api repo?

**Comment by Brutus5000** [comment]
I tried but migrated everything to Lombok builders, but then encountered that MapStruct doesn't support builders in our scenario (confirmed by the maintainer).

**Review by Brutus5000 — APPROVED** [review]

**Pull request merged**